ChiTech/ChiMesh/SweepUtilities/sweepchunk_base.h

```diff
@@ -40,6 +40,17 @@ class chi_mesh::sweep_management::SweepChunk
     x = (&destination_phi);
   }
 
+  /**Sets all elements of the output vector to zero.*/
+  void ZeroDestinationPhi()
+  {
+    (*x).assign((*x).size(), 0.0);
+  }
+
+  virtual void ZeroDestinationPsi() = 0;
+  virtual void ZeroIncomingDelayedPsi() = 0;
+  virtual void ZeroOutgoingDelayedPsi() = 0;
+  virtual void ZeroFluxDataStructures() = 0;
```

Comment:
With the present architecture of the base class `chi_mesh::sweep_management::SweepChunk` and its derived classes, it is not ideal to declare these last four `Zero...` methods as pure virtual in this location. In particular, the methods modify class member variables that do not exist until defined in a derived class. I recommend to treat these methods in a manner analogous to get/set methods: declare and define them in the same class that defines the member variable that is modified by the method. An additional drawback of the present layout is that a derived class which does not employ the concept of a stored angular flux must override these methods nevertheless.

Two ideas come to mind:
1. move the `psi_new_local` and `groupset` (and perhaps other) member variables to the class `chi_mesh::sweep_management::SweepChunk` and define the `Zero...` methods there;
2. move the declaration of the `Zero...` methods to the class `LinearBoltzmann::SweepChunkPWL`; this may require a minor refactor of LBS to work with the derived class.

I believe that the intention of `chi_mesh::sweep_management::SweepChunk` is to be quite abstract, so likely the second option. @Naktakala, @wdhawkins?

Reply:
I had this same thought and discussed it with @Naktakala. This implementation was his recommendation.  Initially, these methods were declared and implemented within `LinearBoltzmann::SweepChunkPWL`, but this led to some issues. Because we pass `chi_mesh::sweep_management::SweepScheduler` to the iterative methods and use the sweep chunk it stores of type `chi_mesh::sweep_management::SweepChunk` for sweeping, the virtual methods were needed. I think it adds clarity to the code to have the sweep chunk preparation routines within the sweep chunk. Unless I am misreading the suggestion for the second option, I am not sure it would work for the aforementioned reasons.